src/hooks/useGetCharDetails.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
+
+export const useGetCharDetails = (character) => {
+  const [charDetails, setCharDetails] = useState([]);
+
+  useEffect(() => {
```

Comment:
In my opinion, this logic is unnecessary; such operations are usually performed during component rendering. The use of useEffect should be limited to specific cases.